tests/console/docker_image.pm

```diff
@@ -7,69 +7,74 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-# Summary: Test installation and running of the docker image for this snapshot
-# Maintainer: Fabian Vogt <fvogt@suse.com>
+# Summary: Test installation and running of the docker image from the registry for this snapshot
+# Maintainer: Pavel Dostál <pdostal@suse.cz>
 
 use base 'consoletest';
 use testapi;
 use utils;
 use strict;
-use version_utils qw(is_leap is_tumbleweed);
+use registration "add_suseconnect_product";
+use version_utils "is_sle";
 
 sub run {
-    select_console 'root-console';
+    select_console "root-console";
 
-    my $version = get_required_var('VERSION');
+    my $version     = get_required_var("VERSION");
+    my $SCC_REGCODE = get_required_var("SCC_REGCODE");
 
-    my $image_name;
-    my $image_path;
-
-    if (is_tumbleweed) {
-        $image_name = 'opensuse/tumbleweed:current';
-        $image_path = '/usr/share/suse-docker-images/native/*-image*.tar.xz';
-
-        # For Tumbleweed, the image is wrapped inside an RPM
-        zypper_call "in docker opensuse-tumbleweed-image";
+    if (script_run("SUSEConnect --status-text") != 0) {
+        assert_script_run("SUSEConnect --cleanup");
+        assert_script_run("SUSEConnect -r $SCC_REGCODE");
+        add_suseconnect_product("sle-module-containers", substr($version, 0, 2));
     }
-    elsif (is_leap('15.0+')) {
-        $image_name = "opensuse-leap-$version:current";
-        $image_path = "/usr/share/suse-docker-images/native/opensuse-leap${version}-image.tar.xz";
 
-        # For Leap, the docker image is the ISO asset
-        my $image_filename = get_required_var('ISO');
-        $image_filename =~ s/^.*\///;
-        my $image_url = autoinst_url("/assets/other/$image_filename");
-        assert_script_run "curl $image_url --create-dirs -o $image_path";
+    my $image_name;
+    if (is_sle("=12-SP3")) {
+        $image_name = "registry.suse.de/suse/sle-12-sp3/update/products/casp30/container/sles12:sp3";
+    }
+    elsif (is_sle("=15")) {
+        $image_name = "registry.suse.de/suse/sle-15/update/cr/images/suse/sle15:current";
     }
     else {
-        die 'Only know about Tumbleweed and Leap 15.0+ docker images';
+        die("This test only works at SLE12SP3 and SLE15.");
```

Comment:
I do not understand your intention here. This broke SLE12SP4 tests, e.g. https://openqa.suse.de/tests/1849296#step/docker_image/4

When you have a check for version equality we can not schedule the test module for SLE12-SP3+ . Could you add support for SLE12-SP4 or not schedule the test module on SLE12-SP4?

Reply:
disk image is supposed to be registered